payment/service.go

```diff
@@ -266,3 +291,109 @@ func (s *Service) IsOrderPaid(orderID uuid.UUID) (bool, error) {
+	debug, err := strconv.ParseBool(os.Getenv("DEBUG"))
+	if err != nil {
+		debug = false
+	}
+	return cors.Handler(cors.Options{
+		Debug:            debug,
+		AllowedOrigins:   strings.Split(os.Getenv("ALLOWED_ORIGINS"), ","),
+		AllowedMethods:   allowedMethods,
+		AllowedHeaders:   []string{"Accept", "Authorization", "Content-Type"},
+		ExposedHeaders:   []string{""},
+		AllowCredentials: false,
+		MaxAge:           300, // Maximum value not ignored by any of major browsers
+	})
+}
+
+// SetupService - setup the payment microservice
+func SetupService(ctx context.Context, r *chi.Mux) (*chi.Mux, context.Context, *Service) {
+	logger, err := appctx.GetLogger(ctx)
+	if err != nil {
+		// no logger, setup
+		ctx, logger = logging.SetupLogger(ctx)
+	}
+
+	// setup the service now
+	db, err := NewWritablePostgres(viper.GetString("datastore"), true, "payment_db")
+	if err != nil {
+		logger.Error().Err(err).Msg("unable to connect to payment db")
+		logger.Panic().Err(err).Msg("unable connect to payment db")
+	}
+	//roDB, err := NewReadOnlyPostgres(viper.GetString("ro-datastore"), false, "payment_ro_db")
+	//if err != nil {
+	//	logger.Panic().Err(err).Msg("unable connect to payment db")
+	//}
+
+	ctx = context.WithValue(ctx, appctx.DatastoreCTXKey, db)
+
+	// TODO: fix to be actually a read only connection
+	ctx = context.WithValue(ctx, appctx.RODatastoreCTXKey, db)
+
+	// add our command line params to context
+	ctx = context.WithValue(ctx, appctx.EnvironmentCTXKey, viper.Get("environment"))
+
+	s, err := InitService(ctx, db)
+	if err != nil {
+		logger.Fatal().Err(err).Msg("failed to initialize payment service")
+	}
+
+	// // setup reputation client
+	// s.repClient, err = reputation.New()
+	// // its okay to not fatally fail if this environment is local and we cant make a rep client
+	// if err != nil && os.Getenv("ENV") != "local" {
+	// 	logger.Fatal().Err(err).Msg("failed to initialize payment service")
+	// }
+
+	// ctx = context.WithValue(ctx, appctx.ReputationClientCTXKey, s.repClient)
+
+	// if feature is enabled, setup the routes
+	if viper.GetBool("payments-feature-flag") {
+		// setup our payment routes
+		r.Route("/v1/orders", func(r chi.Router) {
+			if os.Getenv("ENV") == "local" {
+				r.Method("OPTIONS", "/", middleware.InstrumentHandler("CreateOrderOptions", corsMiddleware([]string{"POST"})(nil)))
+				r.Method("POST", "/", middleware.InstrumentHandler("CreateOrder", corsMiddleware([]string{"POST"})(CreateOrder(s))))
+			} else {
+				r.Method("POST", "/", middleware.InstrumentHandler("CreateOrder", CreateOrder(s)))
+			}
+
+			r.Method("OPTIONS", "/{orderID}", middleware.InstrumentHandler("GetOrderOptions", corsMiddleware([]string{"GET"})(nil)))
+			r.Method("GET", "/{orderID}", middleware.InstrumentHandler("GetOrder", corsMiddleware([]string{"GET"})(GetOrder(s))))
+
+			if os.Getenv("ENV") != "production" {
+				r.Method("PUT", "/{orderID}", middleware.InstrumentHandler("CancelOrder", CancelOrder(s)))
```

Comment:
there's no authentication on this. I believe the intended use is to be only used by the merchant ( which in premium case is the Brave subscriptions server ). this is similar to the delete credentials endpoint, both should use merchant specific auth